six/modules/c++/six.sidd/include/six/sidd/SFA.h

```diff
@@ -205,8 +201,10 @@ struct SFAPolyhedralSurface : public SFASurface
         SFASurface(TYPE_NAME)
     {
     }
-    virtual ~SFAPolyhedralSurface()
+
+    virtual SFAPolyhedralSurface* clone() const
     {
+        return new SFAPolyhedralSurface(*this);
     }
 
     std::vector<mem::ScopedCopyablePtr<SFAPolygon> > patches;
```

Comment:
Hmm, so I think you're right.  I'd have to go dig through some C++ documentation to say for sure.  But just for code clarity, let's say any time something is a base class where things are inheriting from it, let's use a vector of cloneables.  I think line 285 is the only remaining one that needs to get changed.